### ISSUE TITLE
SSR support?

### ISSUE THREAD

**Issue by cwirz** [pagekit/vue-resource#455]
I'm trying to use vue-resource together with vuejs 2.0 and `server side rendering`.
Is this possible now?
I'm getting this error:

```
ReferenceError: document is not defined
    at Object.<anonymous> (/usr/src/app/node_modules/vue-resource/dist/vue-resource.common.js:641:10)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (__vue_ssr_bundle__:3489:18)
    at __webpack_require__ (__vue_ssr_bundle__:21:30)
```

**Comment by afrianjunior** [comment]
we have same problem :'D

**Comment by afrianjunior** [comment]
i use this for a while

``` javascript
  import Vue from 'vue'
  const inBrowser = typeof window !== 'undefined'

  if (inBrowser) {
    Vue.use(require('vue-resource'))
  }
```

**Comment by Wirzi** [comment]
That will work but don't fix the problem because you want to fetch the data using a external api not only in the browser but also on the server to make sure ssr is working. I ended up using the [request package](https://www.npmjs.com/package/request), the [request-promise](https://www.npmjs.com/package/request-promise) package and the  [hackernews 2.0](https://github.com/vuejs/vue-hackernews-2.0) as example to get started. Here is my api file as replacement for the hackernews project api.js file
```
import request from 'request-promise'
import format from 'string-format'

export function fetchIdsByType (type) {
    return request('http://hungerstream.gourmendo.com/de/api/food/list-ids/?format=json')
}

export function fetchItem (id) {
    return request(format('http://hungerstream.gourmendo.com/de/api/food/{}/?format=json', id)).then(response => {
        return JSON.parse(response)
    })
}

export function fetchItems (ids) {
    return Promise.all(
        ids.map(id => fetchItem(id))
    )
}
```

**Comment by andreiglingeanu** [comment]
One more idea is to use an HTTP client that's built to be used both on server and in browser, like [axios](https://github.com/mzabriskie/axios).

**Comment by jdrydn** [comment]
Well, I'm glad this is an actual issue and not just me going mad!

**Comment by MarkPieszak** [comment]
Any update on this one?

**Comment by meyagen** [comment]
+1

In case anyone discovers this from google and wants an alternative solution using the fetch API, you can also use the [isomorphic-fetch](https://github.com/matthew-andrews/isomorphic-fetch) polyfill. Note that you'll need to use a promise polyfill also (try [es6-promise](https://github.com/stefanpenner/es6-promise)).

As with @cwirz, here's an example replacing the `api.js` from [vue-hackernews 2.0](https://github.com/vuejs/vue-hackernews-2.0)
```
import 'es6-promise';
import fetch from 'isomorphic-fetch';

const fetchResource = (resource) => {
  return fetch(`https://jsonplaceholder.typicode.com/${resource}`)
  .then((response) => {
    if (!response.ok) {
      console.log(`Looks like there was a problem. Status Code: ${response.status}`);
    }
    
    return response;
  })

  .catch((err) => {
    console.log('Fetch Error :-S', err);
    return err;
  });
}

const fetchPosts = () => (fetchResource('posts'));
const fetchUsers = () => (fetchResource('users'));

export { fetchPosts, fetchUsers };
```

**Comment by jdrydn** [comment]
A month or so later, and I totally agree with @andreiglingeanu: [axios](https://github.com/mzabriskie/axios) solves this problem perfectly, and it's API is excellent to use!

**Comment by steffans** [comment]
Fixed in v1.1.2
